src/app/api/scheduling.api.ts

```diff
@@ -0,0 +1,102 @@
+import { MutationResult } from "apollo-angular";
+import { Injectable } from "@angular/core";
+import {
+    GetDatasetBatchingGQL,
+    GetDatasetBatchingQuery,
+    GetDatasetIngestScheduleGQL,
+    GetDatasetIngestScheduleQuery,
+    ScheduleInput,
+    SetConfigBatchingGQL,
+    SetConfigBatchingMutation,
+    SetConfigScheduleGQL,
+    SetConfigScheduleMutation,
+    TimeDeltaInput,
+} from "./kamu.graphql.interface";
+import { Observable, first, map } from "rxjs";
+import { ApolloQueryResult } from "@apollo/client";
+import { DatasetOperationError } from "../common/errors";
+
+@Injectable({ providedIn: "root" })
+export class SchedulingApi {
```

Comment:
Similar API issues propagate further. Not enough abstraction. Parameterize flow type, we don't need an API method for every type of flow. We need universal API that gets/sets a schedule and gets/sets batching conditions.

Reply:
Fixed. renamed